server/routes/course.js

```diff
@@ -10,114 +10,113 @@ const User = require('./../schemas/user');
 //CREATE course route: Adds a new course as a subdocument to 
 //a document in the users collection (POST)
 router.post('/courses/:userId', async (req, res, next) => {
-    console.log("in /courses (POST) route with params = " + 
-                JSON.stringify(req.params) + " and body = " + 
-                JSON.stringify(req.body));
-    if (!req.body.hasOwnProperty("courseInstructorFirstName") || 
-        !req.body.hasOwnProperty("courseInstructorLastName") || 
-        !req.body.hasOwnProperty("courseInstructorID") ||
-        !req.body.hasOwnProperty("courseName") ||
-        !req.body.hasOwnProperty("courseNumber") || 
-        !req.body.hasOwnProperty("courseYear") ||
-        !req.body.hasOwnProperty("courseSemester") ||
-        !req.body.hasOwnProperty("courseEnrollmentLimit") || 
-        !req.body.hasOwnProperty("courseCurrentlyEnrolled") || 
-        !req.body.hasOwnProperty("courseNotes")) {
-      //Body does not contain correct properties
-      return res.status(400).send("POST request on /courses formulated incorrectly." +
-        "Body must contain all 9 required fields: courseInstructorFirstName, courseInstructorLastName, courseInstructorID, courseName, courseNumber, courseYear, courseYear, courseSemester, courseEnrollmentLimit, courseCurrentlyEnrolled, courseNotes");
+  console.log("in /courses (POST) route with params = " + 
+              JSON.stringify(req.params) + " and body = " + 
+              JSON.stringify(req.body));
+  if (!req.body.hasOwnProperty("courseInstructorFirstName") || 
+      !req.body.hasOwnProperty("courseInstructorLastName") || 
+      !req.body.hasOwnProperty("courseInstructorID") ||
+      !req.body.hasOwnProperty("courseName") ||
+      !req.body.hasOwnProperty("courseNumber") || 
+      !req.body.hasOwnProperty("courseYear") ||
+      !req.body.hasOwnProperty("courseSemester") ||
+      !req.body.hasOwnProperty("courseEnrollmentLimit") || 
+      !req.body.hasOwnProperty("courseCurrentlyEnrolled") || 
+      !req.body.hasOwnProperty("courseNotes")) {
+    //Body does not contain correct properties
+    return res.status(400).send("POST request on /courses formulated incorrectly." +
+      "Body must contain all 10 required fields: courseInstructorFirstName, courseInstructorLastName, courseInstructorID, courseName, courseNumber, courseYear, courseSemester, courseEnrollmentLimit, courseCurrentlyEnrolled, courseNotes");
+  }
+  try {
+    let status = await User.updateOne(
+    {id: req.params.userId},
+    {$push: {courses: req.body}});
```

Comment:
Actually @marcoarceo as I was working on students CRUD, I realized that we need to add a courseId to the body before we push it. Could you add code for this? courseId could be a combo of course number, year and semester.

Reply:
Yes, I can add the courseID by tomorrow morning. We would have to set the courseID after the user adds the information required to create a course. Before calling the route to POST, we would assign courseID to a combination of courseName + courseNumber + courseSemester + courseYear.

Reply:
Sounds good! No rush, I can manually add it in the meantime

Reply:
Added the courseID to the body, please review the code and let me know if that was what you asked for.